### PR TITLE
feat(spans): Normalization for span ingestion

### PR DESCRIPTION
**Pull request by jjbayer**
Run full normalization before extracting metrics from a standalone span.

#skip-changelog

### CODE REVIEW

File: tests/integration/test_store.py [hunk 1]
```diff
@@ -1340,8 +1343,8 @@ def test_span_ingestion(
                         "traceId": "89143b0763095bd9c9955e8175d1fb23",
                         "spanId": "e342abb1214ca181",
                         "name": "my 1st OTel span",
-                        "startTimeUnixNano": 1697620454980000000,
-                        "endTimeUnixNano": 1697620454980078800,
+                        "startTimeUnixNano": int(start.timestamp() * 1_000_000_000),
+                        "endTimeUnixNano": int(end.timestamp() * 1_000_000_000),
```
Comment (jjbayer) [inline tests/integration/test_store.py:1348]: Our clock drift processor fixes very old timestamps, so we have to use dynamic timestamps here.

File: relay-server/src/actors/processor.rs [hunk 2]
```diff
@@ -2110,6 +2112,7 @@ impl EnvelopeProcessorService {
 
     #[cfg(feature = "processing")]
     fn extract_spans(&self, state: &mut ProcessEnvelopeState) {
+        use relay_event_normalization::span::tag_extraction;
```
Comment (olksdr) [inline relay-server/src/actors/processor.rs]: This should be moved to above into the 

```rust
#[cfg(feature = "processing")]
use {
```

File: relay-server/src/actors/processor.rs [hunk 3]
```diff
@@ -2279,59 +2282,95 @@ impl EnvelopeProcessorService {
 
     #[cfg(feature = "processing")]
     fn process_spans(&self, state: &mut ProcessEnvelopeState) {
+        use crate::actors::processor::span::{normalize_span, NormalizeSpanConfig};
```
Comment (olksdr) [inline relay-server/src/actors/processor.rs]: Same here for this 2 imports

File: relay-server/src/actors/processor.rs [hunk 4]
```diff
@@ -2279,59 +2284,92 @@ impl EnvelopeProcessorService {
...
+                    .aggregator_config_for(MetricNamespace::Spans)
+                    .max_tag_value_length,
+            };
+            let mut annotated_span = match normalize_span(annotated_span, config) {
+                Ok(s) => s,
                 Err(e) => {
-                    relay_log::error!("invalid span: {e}");
-                    return;
+                    relay_log::debug!("failed to normalize span: {}", e);
+                    return ItemAction::DropSilently;
```
Comment (iker-barriocanal) [inline relay-server/src/actors/processor.rs]: Why the error -> debug downgrade, considering that spans are silently dropped?

File: relay-server/src/actors/processor.rs [hunk 5]
```diff
@@ -2279,59 +2284,92 @@ impl EnvelopeProcessorService {
...
-                let mut item = Item::new(ItemType::Span);
-                item.set_payload(ContentType::Json, payload);
-                items.push(item);
+                let metrics = extract_metrics(span, config);
+                state.extracted_metrics.project_metrics.extend(metrics);
             }
-        };
-        let standalone_span_ingestion_enabled = state
-            .project_state
-            .has_feature(Feature::StandaloneSpanIngestion);
```
Comment (iker-barriocanal) [inline relay-server/src/actors/processor.rs]: Is the feature flag no longer useful? Without this check, we'll proceed with `ItemType::Span`s too.

File: relay-server/src/actors/processor.rs (hunk 4 above)
Comment (jjbayer) [reply relay-server/src/actors/processor.rs]: Good question! `normalize_span` returns an error when the span is invalid (e.g. missing trace ID), so we don't want to be notified about that. I changed the wording to make the failure case more clear.

File: relay-server/src/actors/processor.rs (hunk 5 above)
Comment (phacops) [reply relay-server/src/actors/processor.rs]: It's already checked in `span::filter` so we didn't think we needed to have it twice.

File: relay-server/src/actors/processor/span.rs [hunk 6]
```diff
@@ -1,18 +1,30 @@
...
-use crate::envelope::ItemType;
-use crate::utils::ItemAction;
-use relay_dynamic_config::Feature;
+use crate::{actors::processor::ProcessEnvelopeState, envelope::ItemType, utils::ItemAction};
+#[cfg(feature = "processing")]
+use {
+    crate::actors::processor::ProcessingError,
+    crate::envelope::{ContentType, Item},
+    crate::metrics_extraction::generic::extract_metrics,
+};
```
Comment (iker-barriocanal) [inline relay-server/src/actors/processor/span.rs:27]: Should we merge the `processing` import blocks into one?

**Review by iker-barriocanal — APPROVED** [review]

File: relay-server/src/actors/processor/span.rs (hunk 6 above)
Comment (jjbayer) [reply relay-server/src/actors/processor/span.rs:27]: We don't have a best practice for this yet. My reasoning was to keep the partition into std lib - other crates - current crate.

**Pull request merged**